src/main/java/com/redhat/rhjmc/containerjfr/commands/internal/PortScanCommand.java

```diff
@@ -48,18 +51,44 @@ public boolean validate(String[] args) {
 
     @Override
     public void execute(String[] args) throws Exception {
-        scan().forEach(m -> cw.println(String.format("%s -> %s", m.hostname, m.ip)));
+        findCompatibleJvms().forEach(m -> cw.println(String.format("%s -> %s", m.hostname, m.ip)));
     }
 
     @Override
     public Output<?> serializableExecute(String[] args) {
         try {
-            return new ListOutput<>(scan());
+            return new ListOutput<>(findCompatibleJvms());
         } catch (Exception e) {
             return new ExceptionOutput(e);
         }
     }
 
+    private List<IpHostMapping> findCompatibleJvms() throws UnknownHostException, InterruptedException {
+        // Check for environment variables added by Kubernetes that indicate
+        // IP addresses listening on port 9091.
+        List<IpHostMapping> mapping = System.getenv().entrySet().parallelStream()
```

Comment:
Please abstract the `System.getenv()` call into `com.redhat.rhjmc.containerjfr.sys.Environment`. This will help eventual testability for this class.

Reply:
Sure, I think I've done so correctly now.